final_task/rss_parser/Output_functions.py

```diff
@@ -286,47 +269,56 @@ def writing_to_file(pack_of_news, filename):
                  If Not in the file: Find out length of file (amount of news), put that number to incoming novelty
     If that novelty exists it will go to another novelty in limit
     It means that if you enter --limit 15 and 10 news are already in list it will add only 5 news to list
-    :param pack_of_news:
-    :param filename:
-    :return:
     """
+    conn = sqlite3.connect("database.db")
+    cursor = conn.cursor()
+    cursor.execute('create table if not exists projects(num integer, title text, time text, source_link text, '
+                   'description text, images_links text, alt_tx text, date_corrected integer, main_source text)')
+    conn.commit()
+    logging.info("Opening file News_cache.")
```

Comment:
Usually it is good practice to actually close connection with the db after you finished fetching/pushing data. You can even use context manager for this.

Reply:
I manually closed connection with db at the end of the function writing_to_file and at the end of function getting_from_database. Am I not right? Or I should/have to/must use context manager for this?

Reply:
It is actually the best practice to use context  manager for this yes. That way it is more compact, even more readable and absolutely safe